base/parse.jl

```diff
@@ -194,3 +194,82 @@ function parse(str::AbstractString; raise::Bool=true)
     end
     return ex
 end
+
+const SPACE   = UInt8(' ')
+const TAB     = UInt8('\t')
+const ZERO    = UInt8('0')
+const NINE    = UInt8('9')
+const littleA = UInt8('a')
+const bigA    = UInt8('A')
+const littleZ = UInt8('z')
+const bigZ    = UInt8('Z')
+
+immutable ParsingError <: Exception
+    msg::String
+end
+
+immutable Options{base}
```

Comment:
this should be an `ImmutableDict`. this type will cause issues for compilation.

Reply:
Can you elaborate? I didn't even realize `ImmutableDict` exists, so it would be interesting and relevant for me to know when I should use it ;)